src/include/OpenImageIO/imagebufalgo_util.h

```diff
@@ -250,15 +250,18 @@ enum IBAprep_flags {
 
 
 
-// DEPRECATED(2.3): Prefer TypeDesc::basetype_merge().
```

Comment:
> Do you remember why a comment was used for the deprecation long ago vs. just adding the real deprecated attribute back then?

Not specifically, but I can guess that it's one of two things:

1. Since the usual progression is marked/considered deprecated -> warn as deprecated -> remove, spread over a few releases, sometimes at the first step I'll put the comment while my memory is clear, so when I revisit it a release or two later to enable the warning, I can just uncomment that line rather than have to do archeology to remind myself exactly when it happened or why. But sometimes I forget to come back after the next release and uncomment it.
2. It may be that I really meant to enable the warnings at some prior time, but found that it was so widely used (within OIIO, or by OSL, or by other users) that it seemed too disruptive to add the warnings at that time and I postponed the task or re-commented it.

But with 3.0 coming soon, my aim really is to make sure that everything that previously warned will be removed, and everything previously considered deprecated gets at least a warning even if it's too inconvenient to remove quite yet. In particular, things that we only decided in 2.5 or 2.6/3.0 itself should be deprecated, I am hesitant to remove entirely with too short a transition. But I am warning and making them inline so that a removal later won't be an ABI break. I will do another pass at this right before 3.0 goes into beta, and "upgrade" a lot of the warnings to removals. But the warnings now will at least let people who do test builds against master to identify the fixes they need before we throw that final switch.

Reply:
We might need some sort of release checklist to go through once a year before bigger version bumps maybe.

Reply:
We have one! (A recent addition.) But I think this is an item I may not have remembered to put on it.